source/backend/api/Services/DispositionFileService.cs

```diff
@@ -117,35 +117,18 @@ public PimsDispositionFile Update(long id, PimsDispositionFile dispositionFile,
                 throw new BusinessRuleViolationException("The file you are editing is not active or draft, so you cannot save changes. Refresh your browser to see file state.");
             }
 
-            ValidateStaff(dispositionFile);
             ValidateVersion(id, dispositionFile.ConcurrencyControlNumber);
 
-            dispositionFile.ThrowContractorRemovedFromTeam(_user, _userRepository);
-
-            var doNotAddToStatuses = new List<string>() { EnumDispositionFileStatusTypeCode.COMPLETE.ToString(), EnumDispositionFileStatusTypeCode.ARCHIVED.ToString(), EnumDispositionFileStatusTypeCode.CANCELLED.ToString(), };
+            // validate disposition file state before proceeding with any database updates
             var currentDispositionFile = _dispositionFileRepository.GetById(id);
+            ValidateFileBeforeUpdate(dispositionFile, currentDispositionFile, userOverrides);
 
-            if (!userOverrides.Contains(UserOverrideCode.DispositionFileFinalStatus) && !doNotAddToStatuses.Contains(currentDispositionFile.DispositionFileStatusTypeCode)
-                && doNotAddToStatuses.Contains(dispositionFile.DispositionFileStatusTypeCode))
-            {
-                throw new UserOverrideException(UserOverrideCode.DispositionFileFinalStatus, "You are changing this file to a non-editable state. (Only system administrators can edit the file when set to Archived, Cancelled or Completed state). Do you wish to continue?");
-            }
-            else if (currentDispositionFile.DispositionFileStatusTypeCode != EnumDispositionFileStatusTypeCode.COMPLETE.ToString()
-                && dispositionFile.DispositionFileStatusTypeCode == EnumDispositionFileStatusTypeCode.COMPLETE.ToString())
-            {
-                if (currentDispositionFile?.PimsDispositionSales?.FirstOrDefault()?.SaleFinalAmt == null)
-                {
-                    throw new BusinessRuleViolationException("You have not added a Sales Price. Please add a Sales Price before completion.");
-                }
-                else if (currentDispositionFile.PimsDispositionFileProperties.Count > 0)
-                {
-                    DisposeOfProperties(dispositionFile, userOverrides);
-                }
-            }
+            var isFileClosing = currentDispositionFile.DispositionFileStatusTypeCode != EnumDispositionFileStatusTypeCode.COMPLETE.ToString() &&
```

Comment:
You could move this assigment bebore calling  private "ValidateFileBeforeUpdate" method, and pass the value to the method as parameter, that way you would make the assigment once.

Reply:
fair- still I feel this is a simple check and perhaps should be kept so the "Validate" doesn't have to trust someone to do the check for itself